GitCommands/Settings/SettingsCache.cs

```diff
@@ -16,19 +16,14 @@ public SettingsCache()
 
         public void Dispose()
         {
-            DisposeImpl();
+            Dispose(true);
             GC.SuppressFinalize(this);
         }
 
-        protected virtual void DisposeImpl()
+        protected virtual void Dispose(bool disposing)
```

Comment:
It is a well known Dispose pattern - https://msdn.microsoft.com/en-us/library/b1yfkh5e%28v=vs.110%29.aspx